### PR TITLE
Refactor ec2_vpc_vpn* modules

### PR DESCRIPTION
**Pull request by alinabuzachis**
##### SUMMARY

Refactor ec2_vpc_vpn* modules

##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- New Module Pull Request

##### COMPONENT NAME

ec2_vpc_vpn
ec2_vpc_vpn_info
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/modules/ec2_vpc_vpn.py [hunk 1]
```diff
@@ -229,199 +243,297 @@
...
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import create_vpn_connection
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import create_vpn_connection_route
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import delete_vpn_connection
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import delete_vpn_connection_route
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import describe_vpn_connections
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import ensure_ec2_tags
 from ansible_collections.amazon.aws.plugins.module_utils.tagging import boto3_tag_list_to_ansible_dict
 from ansible_collections.amazon.aws.plugins.module_utils.tagging import compare_aws_tags
 
 from ansible_collections.community.aws.plugins.module_utils.modules import AnsibleCommunityAWSModule as AnsibleAWSModule
```
Comment (GomathiselviS) [inline plugins/modules/ec2_vpc_vpn.py:473]:
```suggestion
from ansible_collections.amazon.aws.plugins.module_utils.modules import AnsibleAWSModule
```

File: tests/integration/targets/ec2_vpc_vpn/tasks/main.yml [hunk 2]
```diff
@@ -133,53 +134,53 @@
...
-  - name: delete vpn connection
-    ec2_vpc_vpn:
+  - name: Delete EC2 VPN Connection
+    community.aws.ec2_vpc_vpn:
       state: absent
       vpn_connection_id: '{{ vpn_id }}'
     register: result
 
-  - assert:
+  - ansible.builtin.assert:
```
Comment (abikouo) [inline tests/integration/targets/ec2_vpc_vpn/tasks/main.yml:154]: A task name is missing

File: tests/integration/targets/ec2_vpc_vpn/tasks/main.yml [hunk 3]
```diff
@@ -133,53 +134,53 @@
...
+  - name: Delete EC2 VPN Connection - idempotency (check)
+    community.aws.ec2_vpc_vpn:
       state: absent
       vpn_connection_id: '{{ vpn_id }}'
     register: result
-    check_mode: True
+    check_mode: true
 
-  - assert:
+  - ansible.builtin.assert:
```
Comment (abikouo) [inline tests/integration/targets/ec2_vpc_vpn/tasks/main.yml:165]: Same as above

File: tests/integration/targets/ec2_vpc_vpn/tasks/main.yml [hunk 4]
```diff
@@ -133,53 +134,53 @@
...
-  - name: delete vpn connection - idempotency
-    ec2_vpc_vpn:
+  - name: Delete EC2 VPN Connection - idempotency
+    community.aws.ec2_vpc_vpn:
       state: absent
       vpn_connection_id: '{{ vpn_id }}'
     register: result
 
-  - assert:
+  - ansible.builtin.assert:
```
Comment (abikouo) [inline tests/integration/targets/ec2_vpc_vpn/tasks/main.yml:175]: Same here

File: plugins/modules/ec2_vpc_vpn.py [hunk 5]
```diff
@@ -229,199 +243,297 @@
...
+  returned: O(state=present)
+  sample:
+    customer_gateway_id: "cgw-1220c87b"
+gateway_association_state:
+  description: The current state of the gateway association.
+  type: str
+  returned: O(state=present)
   sample:
-    customer_gateway_id: cgw-1220c87b
+    gateway_association_state: "associated"
```
Comment (abikouo) [inline plugins/modules/ec2_vpc_vpn.py:259]:
```suggestion
    "associated"
```

File: plugins/modules/ec2_vpc_vpn.py [hunk 6]
```diff
@@ -229,199 +243,297 @@
...
-  returned: I(state=present)
+  returned: O(state=present)
   type: str
 customer_gateway_id:
   description: The customer gateway connected via the connection.
   type: str
-  returned: I(state=present)
+  returned: O(state=present)
+  sample:
+    customer_gateway_id: "cgw-1220c87b"
```
Comment (abikouo) [inline plugins/modules/ec2_vpc_vpn.py:253]:
```suggestion
    "cgw-1220c87b"
```

File: plugins/modules/ec2_vpc_vpn.py [hunk 7]
```diff
@@ -229,199 +243,297 @@
...
-    customer_gateway_id: cgw-1220c87b
+    gateway_association_state: "associated"
 vpn_gateway_id:
   description: The virtual private gateway connected via the connection.
   type: str
-  returned: I(state=present)
+  returned: O(state=present)
   sample:
-    vpn_gateway_id: vgw-cb0ae2a2
+    vpn_gateway_id: "vgw-cb0ae2a2"
```
Comment (abikouo) [inline plugins/modules/ec2_vpc_vpn.py:265]:
```suggestion
    "vgw-cb0ae2a2"
```

File: plugins/modules/ec2_vpc_vpn.py [hunk 8]
```diff
@@ -229,199 +243,297 @@
...
 transit_gateway_id:
   description: The transit gateway id to which the vpn connection can be attached.
   type: str
-  returned: I(state=present)
+  returned: O(state=present)
   sample:
-    transit_gateway_id: tgw-cb0ae2a2
+    transit_gateway_id: "tgw-cb0ae2a2"
```
Comment (abikouo) [inline plugins/modules/ec2_vpc_vpn.py:271]:
```suggestion
    "tgw-cb0ae2a2"
```

File: plugins/modules/ec2_vpc_vpn.py [hunk 9]
```diff
@@ -229,199 +243,297 @@
...
+  returned: O(state=present)
   contains:
     static_routes_only:
       description: If the VPN connection only allows static routes.
-      returned: I(state=present)
-      type: str
+      returned: O(state=present)
+      type: bool
       sample:
         static_routes_only: true
```
Comment (abikouo) [inline plugins/modules/ec2_vpc_vpn.py:283]:
```suggestion
        true
```

File: plugins/modules/ec2_vpc_vpn.py [hunk 10]
```diff
@@ -229,199 +243,297 @@
...
+    enable_acceleration:
+      description: Indicates whether acceleration is enabled for the VPN connection.
+      returned: O(state=present)
+      type: bool
+      sample:
+        enable_acceleration: false
```
Comment (abikouo) [inline plugins/modules/ec2_vpc_vpn.py:289]:
```suggestion
      sample: false
```

File: plugins/modules/ec2_vpc_vpn.py [hunk 11]
```diff
@@ -229,199 +243,297 @@
...
+    local_ipv4_network_cidr:
+      description: The IPv4 CIDR on the customer gateway (on-premises) side of the VPN connection.
+      returned: O(state=present)
+      type: str
+      sample:
+        local_ipv4_network_cidr: "0.0.0.0/0"
```
Comment (abikouo) [inline plugins/modules/ec2_vpc_vpn.py:295]:
```suggestion
      sample: "0.0.0.0/0"
```

File: plugins/modules/ec2_vpc_vpn.py [hunk 12]
```diff
@@ -229,199 +243,297 @@
...
+    outside_ip_address_type:
+      description: The external IP address of the VPN tunnel.
+      returned: O(state=present)
+      type: str
+      sample:
+        outside_ip_address_type: "PublicIpv4"
```
Comment (abikouo) [inline plugins/modules/ec2_vpc_vpn.py:301]:
```suggestion
      sample: "PublicIpv4"
```

File: plugins/modules/ec2_vpc_vpn.py [hunk 13]
```diff
@@ -229,199 +243,297 @@
...
+    remote_ipv4_network_cidr:
+      description: The IPv4 CIDR on the Amazon Web Services side of the VPN connection.
+      returned: O(state=present)
+      type: str
+      sample:
+        remote_ipv4_network_cidr: "0.0.0.0/0"
```
Comment (abikouo) [inline plugins/modules/ec2_vpc_vpn.py:307]:
```suggestion
      sample: "0.0.0.0/0"
```

File: plugins/modules/ec2_vpc_vpn.py [hunk 14]
```diff
@@ -229,199 +243,297 @@
...
+    tunnel_inside_ip_version:
+      description: Indicates whether the VPN tunnels process IPv4 or IPv6 traffic.
+      returned: O(state=present)
+      type: str
+      sample:
+        tunnel_inside_ip_version: "ipv4"
```
Comment (abikouo) [inline plugins/modules/ec2_vpc_vpn.py:313]:
```suggestion
      sample: "ipv4"
```

File: plugins/modules/ec2_vpc_vpn.py [hunk 15]
```diff
@@ -229,199 +243,297 @@
...
+      type: str
+      returned: O(state=present)
 state:
   description: The status of the VPN connection.
   type: str
-  returned: I(state=present)
+  returned: O(state=present)
   sample:
-    state: available
+    state: "available"
```
Comment (abikouo) [inline plugins/modules/ec2_vpc_vpn.py:384]:
```suggestion
   "available"
```

File: plugins/modules/ec2_vpc_vpn.py [hunk 16]
```diff
@@ -229,199 +243,297 @@
...
 tags:
   description: The tags associated with the connection.
   type: dict
-  returned: I(state=present)
+  returned:O(state=present)
   sample:
     tags:
```
Comment (abikouo) [inline plugins/modules/ec2_vpc_vpn.py:390]:
```suggestion
```

File: plugins/modules/ec2_vpc_vpn.py [hunk 17]
```diff
@@ -229,199 +243,297 @@
...
-      name: ansible-test
-      other: tag
+      name: "ansible-test"
+      other: "tag"
```
Comment (abikouo) [inline plugins/modules/ec2_vpc_vpn.py:392]:
```suggestion
{
      name: "ansible-test",
      other: "tag"
}
```

File: plugins/modules/ec2_vpc_vpn.py [hunk 18]
```diff
@@ -229,199 +243,297 @@
...
 type:
   description: The type of VPN connection (currently only ipsec.1 is available).
   type: str
-  returned: I(state=present)
+  returned: V(state=present)
   sample:
     type: "ipsec.1"
```
Comment (abikouo) [inline plugins/modules/ec2_vpc_vpn.py:398]:
```suggestion
    "ipsec.1"
```

File: plugins/modules/ec2_vpc_vpn.py [hunk 19]
```diff
@@ -229,199 +243,297 @@
...
+      type: str
+      sample: "arn:aws:acm:us-east-1:123456789012:certificate/c544d8ce-20b8-4fff-98b0-example"
 vpn_connection_id:
   description: The identifier for the VPN connection.
   type: str
-  returned: I(state=present)
+  returned: O(state=present)
   sample:
-    vpn_connection_id: vpn-781e0e19
+    vpn_connection_id: "vpn-781e0e19"
```
Comment (abikouo) [inline plugins/modules/ec2_vpc_vpn.py:444]:
```suggestion
   "vpn-781e0e19"
```

File: plugins/modules/ec2_vpc_vpn.py (hunk 1 above)
Comment (abikouo) [reply plugins/modules/ec2_vpc_vpn.py:473]: This has to be updated once the module is migrated to `amazon.aws`. Until it lives there this should not be changed

File: plugins/modules/ec2_vpc_vpn.py [hunk 20]
```diff
@@ -638,18 +733,18 @@ def check_for_update(connection, module_params, vpn_connection_id):
         if attribute in ("tags", "routes", "state"):
             continue
         elif attribute == "options":
-            will_be = module_params.get("static_only", None)
+            will_be = module.params.get("static_only", None)
```
Comment (abikouo) [inline plugins/modules/ec2_vpc_vpn.py:736]:
```suggestion
            will_be = module.params.get("static_only")
```

File: plugins/modules/ec2_vpc_vpn.py [hunk 21]
```diff
@@ -638,18 +733,18 @@ def check_for_update(connection, module_params, vpn_connection_id):
...
             is_now = bool(current_attrs[attribute]["static_routes_only"])
             attribute = "static_only"
         elif attribute == "type":
-            will_be = module_params.get("connection_type", None)
+            will_be = module.params.get("connection_type", None)
```
Comment (abikouo) [inline plugins/modules/ec2_vpc_vpn.py:740]:
```suggestion
            will_be = module.params.get("connection_type")
```

File: plugins/modules/ec2_vpc_vpn.py [hunk 22]
```diff
@@ -638,18 +733,18 @@ def check_for_update(connection, module_params, vpn_connection_id):
...
             is_now = current_attrs[attribute]
         else:
             is_now = current_attrs[attribute]
-            will_be = module_params.get(attribute, None)
+            will_be = module.params.get(attribute, None)
```
Comment (abikouo) [inline plugins/modules/ec2_vpc_vpn.py:744]:
```suggestion
            will_be = module.params.get(attribute)
```

**Review by abikouo — CHANGES_REQUESTED** [review]

File: plugins/modules/ec2_vpc_vpn.py (hunk 1 above)
Comment (GomathiselviS) [reply plugins/modules/ec2_vpc_vpn.py:453]: @abikouo Is there a reason for waiting until migration?

Comment (abikouo) [reply plugins/modules/ec2_vpc_vpn.py:453]: @GomathiselviS yes. Each module should use the corresponding class from its collection.
This is needed for the AWS agent which is set into AWS requests. The agent contains the name of the collection from which the module is executed.

**Review by abikouo — APPROVED** [review]

**Review by GomathiselviS — APPROVED** [review]

**Comment by alinabuzachis** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

**Pull request merged**